pkgs/by-name/li/limine/package.nix

```diff
@@ -0,0 +1,61 @@
+# Builds limine with all available features.
+
+{
+  # Helpers
+  fetchurl
+, lib
+, stdenvNoCC
+, # Dependencies
+  llvmPackages
+, gnumake
+, mtools
+, nasm
+}:
+
+let
+  version = "7.1.0";
+  # It is recommended by limine to use the release tarballs instead the git repo.
+  limineSrc = fetchurl {
+    url = "https://github.com/limine-bootloader/limine/releases/download/v${version}/limine-${version}.tar.gz";
+    sha256 = "sha256-oaQA461MGVuG7aJri2HyYb6Pu8BXOWTF3Wq/SputQSw=";
+  };
+in
+stdenvNoCC.mkDerivation {
```

Comment:
Make this attrset `rec` so that `version` becomes accessible to `src`.

Reply:
very-nit: At least nix.dev [recommends against rec](https://nix.dev/guides/best-practices#recursive-attribute-set-rec). I've seen this pattern a lot recently:

```nix
stdenv.mkDerivation (finalAttrs: {
  src = fetchFromGitHub {
    owner = "falkTX";
    repo = finalAttrs.pname;
    rev = "v${finalAttrs.version}";
    hash = "sha256-H15T/z/IRfgWdqToTzq2eJ7q3n9Kj44IZXsd4uaipuU=";
  };
  # ...
```

Not sure this is much better. The simple `let` is pretty ok in my view, because it's easy to see what happens (also when overriding it).